hadoop-ssm-project/src/main/java/org/apache/hadoop/ssm/mover/MoverPool.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ssm.mover;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hdfs.HdfsConfiguration;
+import org.apache.hadoop.util.ToolRunner;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * MoverPool : A singleton class to manage all Mover actions.
+ */
+public class MoverPool {
+  private static MoverPool instance = new MoverPool();
+  private static Configuration conf = new HdfsConfiguration();
+
+  public static MoverPool getInstance() {
+    return instance;
+  }
+
+  private Map<UUID, Status> moverMap;
+
+  private MoverPool() {
+    moverMap = new ConcurrentHashMap<>();
+  }
+
+  /**
+   * Initialize the HDFS configuration of MoverPool at the beginning of
+   * the SSM service.
+   * @param configuration
+   */
+  public static void init(Configuration configuration) {
+    conf = configuration;
+  }
+
+  /**
+   * Create a Mover event.
+   * @param dir the directory to enforce the storage policy using Mover tool.
+   * @return the UUID of this action for user to track status
+   */
+  public UUID createMoverAction(String dir) {
```

Comment:
just a reminder, an move action can also be applied on a single file, so parameter 'dir' may cause some misunderstanding.